contrib/babelfishpg_tsql/src/catalog.c

```diff
@@ -4977,6 +4981,36 @@ rename_tsql_db(char *old_db_name, char *new_db_name)
 		CommitTransactionCommand();
 }
 
+/*
+ * Returns true if the user/role exists in the sys.babelfish_authid_user_ext catalog,
+ * false otherwise.
+ */
+bool
+user_exists_for_db(const char *db_name, const char *user_name)
+{
+	HeapTuple		tuple_cache;
+	NameData		rolname;
+	bool			user_exists = false;
+
+	namestrcpy(&rolname, user_name);
+
+	tuple_cache = SearchSysCache1(AUTHIDUSEREXTROLENAME, NameGetDatum(&rolname));
+
+	if (HeapTupleIsValid(tuple_cache))
+	{
+		bool isnull;
+		char *db_name_from_cache = TextDatumGetCString(SysCacheGetAttr(AUTHIDUSEREXTROLENAME, tuple_cache,
+												 Anum_bbf_authid_user_ext_database_name, &isnull));
+
+		if (strcmp(db_name_from_cache, db_name) == 0)
```

Comment:
we are assuming that db_name_from_cache will not be NULL, is that valid for all the cases?

Reply:
`Assert(!isnull);`

Reply:
Done